cpp/src/parquet.cpp

```diff
@@ -124,11 +146,77 @@ struct create_result_store_fn {
 
 }  // namespace
 
+// Arrow wants column indices instead of names
+// Find the columns
+std::vector<int> GetArrowColumnIndices(const std::string& file_path,
+                                       const std::vector<std::string>& columns)
+{
+  // Open the file and get the schema
+  auto input = ARROW_RESULT(arrow::io::ReadableFile::Open(file_path));
+  std::unique_ptr<parquet::arrow::FileReader> arrow_reader;
+  auto status = parquet::arrow::OpenFile(input, arrow::default_memory_pool(), &arrow_reader);
+  std::shared_ptr<arrow::Schema> schema;
+  status = arrow_reader->GetSchema(&schema);
+
+  // Find the column indices
+  std::vector<int> column_indices;
+  column_indices.reserve(columns.size());
+  for (const auto& col : columns) {
+    auto idx = schema->GetFieldIndex(col);
+    if (idx < 0) { throw std::runtime_error("Column " + col + " not found in Parquet schema."); }
+    column_indices.push_back(idx);
+  }
+  return column_indices;
+}
```

Comment:
Should we maybe just add that information in the prep step already and pass it in even if the gpu task ends up not using it? But this seems OK also and includes automatically raising an error on mismatch.